Fix aria attributes

maier49 opened the pull request:
**Type:** bug

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**
The changes to aria attribute types for dojo/framework#735 caused some type errors here and this fixes them.

tomdye commented on src/button/index.tsx:
```diff
@@ -80,7 +80,7 @@ export const Button = factory(function Button({
 			focus={focus.shouldFocus()}
 			name={name}
-			type={type}
+			type={type as any}
```
I'm not sure that this should be an `as any`?

maier49 replied:
Yeah I'll change it to `as 'submit' | 'reset' | 'button'`. Should have commented on this but I was wondering what we thought about whether the `'menu'` type should be adopted into the framework level or if we it is correct to treat this as an exception considering there's no browser support for `type="menu"`.

tomdye replied:
why does it need to be cast at all? is the `property` not typed already? regarding `menu`, is it a property we currently support? if not, we should add it to our button type typings

tomdye commented on src/calendar/tests/unit/DatePicker.spec.tsx:
```diff
@@ -115,7 +115,7 @@ const expectedMonthPopup = function(open: boolean) {
 			id=""
 			key="month-grid"
-			aria-hidden={`${!open}`}
+			aria-hidden={!open ? 'true' : 'false'}
```
could this ternary be reversed so that you don't use the negative?

tomdye commented on src/calendar/tests/unit/DatePicker.spec.tsx:
```diff
@@ -138,7 +138,7 @@ const expectedYearPopup = function(
 		<div
 			key="year-grid"
-			aria-hidden={`${!open}`}
+			aria-hidden={!open ? 'true' : 'false'}
```
same comment re. negative

tomdye commented on src/range-slider/index.tsx:
```diff
@@ -155,9 +155,9 @@ export const RangeSlider = factory(function RangeSlider({
 		...formatAriaProperties(aria),
 		'aria-describedby': isSlider1 ? minLabelId : maxLabelId,
-		'aria-invalid': valid === false ? 'true' : null,
+		'aria-invalid': (valid === false ? 'true' : undefined) as 'true' | undefined,
```
is the `as` needed here?

maier49 replied:
Unfortunately, since TS doesn't infer the correct type here. I've changed it to use the exported `AriaAttributes` interface explicitly though.

tomdye commented on src/tab-container/index.tsx:
```diff
@@ -181,7 +181,7 @@ export const TabContainer = factory(function TabContainer({
 
 	let alignClass;
-	let orientation = 'horizontal';
+	let orientation: 'horizontal' | 'vertical' = 'horizontal';
```
Should we create a type for this?

maier49 replied:
I changed it to use the existing property on the `AriaAttributes` interface

tomdye commented on src/tab-container/tests/TabContainer.spec.tsx:
```diff
@@ -61,8 +61,8 @@ const reverseOrientationTemplate = assertionTemplate(() => (
 	'aria-controls': 'test-tab-0',
-	'aria-disabled': 'false',
-	'aria-selected': 'true',
+	'aria-disabled': 'false' as 'false',
```
is this really necessary? Why does `false` need to be cast to `false`?

maier49 replied:
Unfortunately, since TS doesn't infer the correct type here. I've changed it to use the exported `AriaAttributes` interface explicitly though.

maier49 commented:
@tomdye If we want to use the `AriaAttributes` interface then this can't be merged until we have a version of framework that includes that. Additionally, if we want to support booleans to avoid the need to cast strings that would require a change to vdom functionality as well as types. So in the meantime I've addressed the feedback and converted this to a draft.

agubler commented:
Superseded by https://github.com/dojo/widgets/pull/1569
